File: strings/rotate_string.rb
# @param {String} s
# @param {String} goal
# @return {Boolean}
def rotate_string(s, goal)
    arr = s.split("")
    goal_arr = goal.split("")
    (arr.size).times do
        arr << arr.shift
        return true if arr == goal_arr
    end
    false
end

File: strings/rotate_string.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        if len(s) != len(goal):
            return False
            
        arr = list(s)
        goal_arr = list(goal)
        for _ in range(len(arr)):
            arr.append(arr.pop(0))
            if arr == goal_arr:
                return True
                
        return False
